### PR TITLE
test: Outstanding List component validation tests

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Complete remaining `test.todo()`s following on from https://github.com/theopensystemslab/planx-new/pull/3217
 - These tests check error handling and form validation within the list component
 - There's an outstanding validation test for checklists, I'll either pick this up here as a subsequent commit, or on https://github.com/theopensystemslab/planx-new/pull/3304 (dependant on merge order)

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/List/Public/Fields.tsx [hunk 1]
```diff
@@ -16,6 +15,7 @@ import { DESCRIPTION_TEXT, ERROR_MESSAGE } from "../../shared/constants";
 import BasicRadio from "../../shared/Radio/BasicRadio";
 import type { NumberField, QuestionField, TextField } from "../model";
 import { useListContext } from "./Context";
+import { get } from "lodash";
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/List/Public/Fields.tsx:18]: This change was required as the formik's `getIn()` doesn't correctly handle `fn` fields using dot notation (e.g. `proposal.propertyType`).

Docs: https://formik.org/docs/api/fieldarray#fieldarray-validation-gotchas

File: editor.planx.uk/src/@planx/components/List/Public/Context.tsx [hunk 2]
```diff
@@ -84,6 +84,11 @@ export const ListProvider: React.FC<ListProviderProps> = (props) => {
     // Do not allow a new item to be added if there's still an active item
     if (activeIndex !== -1) return setAddItemError(true);
 
+    // Do not allow new item to be added if it will exceed max
+    if (schema.max && formik.values.userData.length === schema.max) {
+      return setMaxError(true);
+    }
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/List/Public/Context.tsx:90]: Moving this here feels like a better user experience than just checking this when a user hits "Continue" (and may have added too many items).

File: editor.planx.uk/src/@planx/components/List/Public/Fields.tsx (hunk 1 above)
Comment (jessicamcinchak) [reply editor.planx.uk/src/@planx/components/List/Public/Fields.tsx:18]: :bulb:

**Review by jessicamcinchak — APPROVED** [review]
Meaningful coverage here :100:

**Pull request merged**